controllers/common-controller-utils.go

```diff
@@ -0,0 +1,352 @@
+package controllers
```

Comment:
@aruniiird Maybe you missed this.

Reply:
Can we do it as a refactor PR later, because most of the functionality is used for the resources in `controllers` package. Moving it to another utility common package requires a quite some API changes.
More than that, most of us have started working with this current controller utilities.